node-ram-cache.cpp

```diff
@@ -281,157 +299,169 @@ int node_ram_cache::get_sparse(osmNode *out, osmid_t id) {
+            }
+        } else {
+            free(blockCache);
+            blockCache = 0;
+        }
+        free(blocks);
+        free(queue);
   }
   if ( ((allocStrategy & ALLOC_SPARSE) > 0) && ((allocStrategy & ALLOC_DENSE) == 0)) {
       free(sparseBlock);
   }
 }
 
-void node_ram_cache::set(osmid_t id, double lat, double lon) {
-    if ((id > 0 && id >> BLOCK_SHIFT >> 32) || (id < 0 && ~id >> BLOCK_SHIFT >> 32 )) {
-        fprintf(stderr, "\nAbsolute node IDs must not be larger than %" PRId64 " (got%" PRId64 " )\n",
-                (int64_t) 1 << 42, (int64_t) id);
+void node_ram_cache::set(osmid_t id, const osmium::Location &coord)
+{
+    if ((id > 0 && id >> BLOCK_SHIFT >> 32) ||
+        (id < 0 && ~id >> BLOCK_SHIFT >> 32)) {
+        fprintf(stderr, "\nAbsolute node IDs must not be larger than %" PRId64
+                        " (got%" PRId64 " )\n",
+                (int64_t)1 << 42, (int64_t)id);
         util::exit_nicely();
     }
     totalNodes++;
     /* if ALLOC_DENSE and ALLOC_SPARSE are set, send it through
-     * ram_nodes_set_dense. If a block is non dense, it will automatically
-     * get pushed to the sparse cache if a block is sparse and ALLOC_SPARSE is set
-     */
-    if ( (allocStrategy & ALLOC_DENSE) > 0 ) {
-        set_dense(id, ramNode(lon, lat));
-    } else if ( (allocStrategy & ALLOC_SPARSE) > 0 ) {
-        set_sparse(id, ramNode(lon, lat));
+   * ram_nodes_set_dense. If a block is non dense, it will automatically
+   * get pushed to the sparse cache if a block is sparse and ALLOC_SPARSE is set
+   */
+    if ((allocStrategy & ALLOC_DENSE) > 0) {
+        set_dense(id, coord);
+    } else if ((allocStrategy & ALLOC_SPARSE) > 0) {
+        set_sparse(id, coord);
     } else {
         // Command line options always have ALLOC_DENSE | ALLOC_SPARSE
-        throw std::logic_error((boost::format("Unexpected cache strategy in node_ram_cache::set with allocStrategy %1%") % allocStrategy).str());
+        throw std::logic_error(
+            (boost::format(
+                 "Unexpected cache strategy in node_ram_cache::set with "
+                 "allocStrategy %1%") %
+             allocStrategy)
+                .str());
     }
 }
 
-int node_ram_cache::get(osmNode *out, osmid_t id) {
-    nodesCacheLookups++;
+osmium::Location node_ram_cache::get(osmid_t id)
+{
+    osmium::Location coord;
 
-    if ((allocStrategy & ALLOC_DENSE) > 0) {
-        if (get_dense(out, id) == 0) {
-            nodesCacheHits++;
-            return 0;
-        }
+    if (allocStrategy & ALLOC_DENSE) {
+        coord = get_dense(id);
     }
-    if ((allocStrategy & ALLOC_SPARSE) > 0) {
-        if (get_sparse(out, id) == 0) {
-            nodesCacheHits++;
-            return 0;
-        }
+
+    if (allocStrategy & ALLOC_SPARSE && !coord.valid()) {
+        coord = get_sparse(id);
     }
 
-    return 1;
+    if (coord.valid()) {
+        nodesCacheHits++;
+    }
+    nodesCacheLookups++;
+
+    return coord;
 }
```

Comment:
If any reformatting has been missed by only doing the diffs, I'd just as soon reformat the entire file. Of course, you might have already thought of this ;)